### DISCUSSION TITLE
Using `crc` for local development of open data hub

### DISCUSSION THREAD

**Discussion by jjasghar** [opendatahub-io/opendatahub-operator#238] [General]
Hey everyone, i was trying to get `crc` running for local development of opendata hub, and help build some content around it. These are the steps followed:

# CRC and OpenDatahub
8 virtual CPUs  (on my laptop it runs at about 50-60% cpu usage)
20GB of memory (on my laptop it takes it all)
70GB of storage space

1. Configure `crc` on your local machine
```bash
crc start
crc config set cpus 8
crc config set memory 20000
crc config set disk-size 70
crc stop
crc start
```
2. Click on Operatorhub
3. Create a namespace called `odh` (confirm it’s active)
4. Search for Open Data Hub
5. Click on the Community version
6. Click “Continue”
7. Click Install
8. Leave defaults, click “Install”
9. Wait till installation is complete
10. Click “View Operator”
11. Change project from “openshift-operators” to “odh”
12. Click “Open Data Hub Operator” if the Status is “Succeeded”
13. Click “Create instance” in the “Kf Def” box
14. Confirm the project is “odh” and click “Create”
15. Click to developer view
16. Click “topology” and watch odh being installed
17. It is completed when `trustyai-service` is has a blue dark ring around it
18. Click https://odh-dashboard-odh.apps-crc.testing/ to get to the dashboard

## Log into the dashboard
1. Click on resources
2. Then step through “Deploying a sample Python application using Flask and OpenShit”
3. Verify it works (it does)

## Starting notebook server
Container size
> Medium fails:
0/1 nodes are available: 1 Insufficient cpu, 1 Insufficient memory. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
> Small fails:
0/1 nodes are available: 1 Insufficient memory. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..

The biggest problem is even scaling up the `crc` instance on my laptop, I still don't have the resources to get the server running. Any advice or direction would be greatly appreciated.

**Comment by oc-will-holtam** [comment]
Hi, yes I have been attempting this as well, so that we can validate that it's appropriate for our requirements before committing to this operator... although I don't think that's possible on my current setup (M2 Macbook Pro 12 CPUS and 16GB memory). According to the OpenDataHub docs, the minimum requirements for the basic tutorials post installation are an OpenShift cluster with a minimum of 16 CPUS and 32GB of memory across all OpenShift worker nodes. 

https://opendatahub.io/docs/quick-installation/
